test/loco/integer/sendmoremoney2.clj

```diff
@@ -0,0 +1,80 @@
+(ns ^{:author "Andrea Richiardi"
+      :doc "Send + More = Money example + tests, second version." }
+  loco.integer.sendmoremoney2
+  (:refer-clojure :exclude [send]) 
+  (:require [clojure.test :refer :all]
+            [loco.core :refer :all]
+            [loco.constraints :refer :all]
+            [criterium.core :refer [bench]]))
+
+(defn costraint-0-9 [vars]
+  (for [v vars]
+    ($in v 0 9)))
+
+(def letters
+  [:S :E :N :D :M :O :R :Y])
+
+(def carries
+  (for [i (range 4)]
+    ($in [:_c i] 0 1)))
+
+(def ^{:doc "Naive model, $distinct on each letter. Constraint on the sum with carries:
+               c3 c2 c1 c0     
+                S  E  N  D +
+                M  O  R  E =
+             ---------------
+             M  O  N  E  Y  "}
+  smm-naive-model
+  (concat (costraint-0-9 letters)
+          carries
+          [($distinct letters)
+           ($!= :S 0)
+           ($!= :M 0)
+           ($= ($+         :D :E) ($+ ($* 10 [:_c 0]) :Y))
+           ($= ($+ [:_c 0] :N :R) ($+ ($* 10 [:_c 1]) :E))
+           ($= ($+ [:_c 1] :E :O) ($+ ($* 10 [:_c 2]) :N))
+           ($= ($+ [:_c 2] :S :M) ($+ ($* 10 [:_c 3]) :O))
+           ($=     [:_c 3]                            :M)]))
+
+(defn reorder-solution
+  "Manages to rearrange a solution in a better way."
+  [solution]
+  (hash-map :send [(find solution :S) (find solution :E) (find solution :N) (find solution :D)]
+            :more [(find solution :M) (find solution :O) (find solution :R) (find solution :E)]
+            :money [(find solution :M) (find solution :O) (find solution :N) (find solution :E) (find solution :Y)]))
+
+(defn random-solution
```

Comment:
I'd like these examples to also serve as test cases to exercise Loco's ability to solve these problems. Part of that exercise is to ensure that the right solutions (and amount of solutions) is returned. Since this problem is a one-solution problem (as far as I know), I'd like the test case to reflect that, since otherwise that would mean that either Loco or the constraint model is incorrect.